### Чтение файлов в Pandas. Практическое задание

В этом практическом задании две обязательные задачи.

*Обязательные задачи* нужно сделать, чтобы проверить, что вы действительно усвоили материал урока. Если у вас всё получилось, вы можете переходить к следующему уроку.

Удачи!

Цели практического задания:

1. Научиться загружать данные из файлов форматов xlsx и csv.
2. Отработать обращение к элементам датафрейма по индексам и названиям строк и столбцов.

В конце ноутбука вы можете найти ответы для самостоятельной проверки.




---


**Обязательные задачи**

Вы будете работать с частью набора рекламных данных, показывающих, щёлкнул ли конкретный пользователь рекламу на сайте компании или нет.

Описание датасета:
- **Number**: Номер пользователя в системе.
- **Daily Time Spent on Site**: Среднее ежедневное время нахождения пользователя на сайте в минутах.
- **Daily Internet Usage**: Среднее ежедневное время нахождения пользователя в интернете в минутах.
- **Ad Topic Line**: Заголовок рекламного объявления.
- **Clicked on Ad**: Кликнул пользователь на рекламу или нет.

Даны файлы двух разных форматов с некоторыми данными для датасета выше:

- df4.xlsx.
- df5.csv.

Загрузите в отдельный датафрейм каждый из них, передав верные параметры там, где требуется:

**Задача 1**

1. Загрузите в df4 данные из xlsx-файла и выведите на экран первые пять строк. Не забудьте указать верный параметр header для отображения корректных заголовков.


In [13]:
# Допишите код
import pandas as pd
df4 = pd.read_excel('./12.2 Данные для урока и практики/12.2 Данные для урока и практики/df4.xlsx', 
                    header=1)


In [14]:
# Ваш код здесь
df4.head()

Number  Daily Time Spent on Site  Daily Internet Usage  \
0       5                     68.95                256.09   
1       6                     80.23                193.77   
2       7                     69.47                236.50   
3       8                     74.15                245.89   
4       9                       NaN                225.58   

                           Ad Topic Line  Clicked on Ad  
0     Cloned 5thgeneration orchestration              0  
1     Monitored national standardization              0  
2       Organic bottom-line service-desk              0  
3  Triple-buffered reciprocal time-frame              0  
4          Robust logistical utilization              0

2. Найдите в датафрейме df4 значение 193.77 и выведите его на экран, используя выбор элементов по индексу.

In [12]:
# Ваш код c iloc здесь
df4.iloc[1, 2]

193.77

**Задача 2**

1. Загрузите в df5 данные из csv-файла и выведите на экран последние десять строк. Не забудьте указать верный разделитель.


In [18]:
# Допишите код
df5 = pd.read_csv('./12.2 Данные для урока и практики/12.2 Данные для урока и практики/df5.csv', sep='|')

In [20]:
# Ваш код здесь
df5.tail(10)

Number  Daily Time Spent on Site  Daily Internet Usage  \
75     105                     41.49                169.18   
76     106                     63.60                235.28   
77     107                     89.91                194.23   
78     108                     68.18                218.17   
79     109                       NaN                202.16   
80     110                     80.49                229.12   
81     111                     72.23                241.03   
82     112                     42.39                150.99   
83     113                       NaN                135.18   
84     114                     74.02                210.54   

                                   Ad Topic Line  Clicked on Ad  
75               Persevering reciprocal firmware              1  
76           Centralized logistical secured line              0  
77          Innovative background conglomeration              0  
78                  Switchable 3rdgeneration hub              0  
79        Polarized 6thgeneration info-mediaries              0  
80                   Balanced heuristic approach              0  
81                 Focused 24hour implementation              0  
82           De-engineered mobile infrastructure              1  
83  Customer-focused upward-trending contingency              1  
84              Operative system-worthy protocol              0

2. Найдите в датафрейме df5 значение `Polarized 6thgeneration info-mediaries` и выведите его на экран, используя выбор элементов по названию.


In [22]:
# Ваш код c loc здесь
df5.loc[79, 'Ad Topic Line']

'Polarized 6thgeneration info-mediaries'



|

|

|

|

|

|

|

|

|

|

|

|

|

|

|

|

|

|

**Ответы для самостоятельной проверки**

**Задача 1**

In [ ]:
# 1.

df4 = pd.read_excel('df4.xlsx', header=1)
df4.head()

Number  Daily Time Spent on Site  Daily Internet Usage  \
0       5                     68.95                256.09   
1       6                     80.23                193.77   
2       7                     69.47                236.50   
3       8                     74.15                245.89   
4       9                       NaN                225.58   

                           Ad Topic Line  Clicked on Ad  
0     Cloned 5thgeneration orchestration              0  
1     Monitored national standardization              0  
2       Organic bottom-line service-desk              0  
3  Triple-buffered reciprocal time-frame              0  
4          Robust logistical utilization              0

In [ ]:
# 2.

df4.iloc[1, 2]

193.77

**Задача 2**

In [ ]:
# 1.

df5 = pd.read_csv('df5.csv', sep='|')
df5.tail(10)

Number  Daily Time Spent on Site  Daily Internet Usage  \
75     105                     41.49                169.18   
76     106                     63.60                235.28   
77     107                     89.91                194.23   
78     108                     68.18                218.17   
79     109                       NaN                202.16   
80     110                     80.49                229.12   
81     111                     72.23                241.03   
82     112                     42.39                150.99   
83     113                       NaN                135.18   
84     114                     74.02                210.54   

                                   Ad Topic Line  Clicked on Ad  
75               Persevering reciprocal firmware              1  
76           Centralized logistical secured line              0  
77          Innovative background conglomeration              0  
78                  Switchable 3rdgeneration hub              0  
79        Polarized 6thgeneration info-mediaries              0  
80                   Balanced heuristic approach              0  
81                 Focused 24hour implementation              0  
82           De-engineered mobile infrastructure              1  
83  Customer-focused upward-trending contingency              1  
84              Operative system-worthy protocol              0

In [ ]:
# 2.

df5.loc[79, 'Ad Topic Line']

'Polarized 6thgeneration info-mediaries'